BraveWallet/Crypto/Stores/TransactionConfirmationStore.swift

```diff
@@ -249,25 +268,26 @@ public class TransactionConfirmationStore: ObservableObject {
     }
   }
   
-  func fetchTransactions() {
-    keyringService.defaultKeyringInfo { [weak self] keyring in
-      guard let self = self else { return }
-      var pendingTransactions: [BraveWallet.TransactionInfo] = []
-      let group = DispatchGroup()
-      for info in keyring.accountInfos {
-        group.enter()
-        self.txService.allTransactionInfo(.eth, from: info.address) { tx in
-          defer { group.leave() }
-          pendingTransactions.append(contentsOf: tx.filter { $0.txStatus == .unapproved })
-        }
-      }
-      group.notify(queue: .main) {
-        self.transactions = pendingTransactions
-        if let firstTx = self.transactions.first { // only do additional set up if we have some unapproved tx
-          self.activeTransactionId = firstTx.id
-          self.fetchGasEstimation1559()
-        }
-      }
+  func prepare() {
+    fetchTransactions { [weak self] in
+      guard let self = self,
+            let firstTx = self.transactions.first
+      else { return }
+      self.activeTransactionId = firstTx.id
+      self.fetchGasEstimation1559()
+    }
+  }
+  
+  func editNonce(
+    for transaction: BraveWallet.TransactionInfo,
+    nonce: String,
+    completion: ((Bool) -> Void)? = nil
```

Comment:
Also doesn't need to be optional, only used in one spot and doesn't alter any mutable Published state in the store, so the only way to actually handle success is via the closure so it should be required at call site